ts/components/staking/remove_stake_dialog.tsx

```diff
@@ -0,0 +1,163 @@
+                </StyledParagraph>
+                <StyledParagraph>
+                    This will take into effect in the next epoch ({stakingUtils.getTimeToEpochDate(nextEpochStart)}).
+                </StyledParagraph>
+                <StyledParagraph>
+                    The rewards you have collected with this pool{' '}
+                    {availableRewardsMap?.[pool.poolId]
+                        ? `(${formatEther(availableRewardsMap[pool.poolId]).minimized} ETH) `
+                        : ''}
+                    will automatically be sent to your wallet now, reflecting in a decrease in the 'available rewards'
+                    balance.
+                </StyledParagraph>
+                <ButtonWrapper>
+                    <ConfirmButton
+                        onClick={() => {
+                            onRemoveStake(pool.poolId, zrxAmount);
+                            onDismiss();
+                        }}
+                    >
+                        I understand, remove my stake
+                    </ConfirmButton>
+                </ButtonWrapper>
+            </StyledDialogContent>
+        </StyledDialogOverlay>
+    );
+};
+
+const ButtonWrapper = styled.div`
+    margin-top: 33px;
+    display: flex;
+    justify-content: flex-end;
+
+    @media (max-width: 768px) {
+        justify-content: center;
+    }
+`;
+
+const StyledParagraph = styled(Paragraph)`
+    font-style: normal;
+    font-weight: 300;
+    font-size: 18px;
+    line-height: 26px;
+`;
+
+const StyledHeading = styled(Heading)`
+    font-size: 34px;
+    line-height: 42px;
+    margin-bottom: 20px;
+
+    font-feature-settings: 'tnum' on, 'lnum' on;
+`;
+
+const ConfirmButton = styled(Button)`
+    background-color: ${props => props.isDisabled && '#898990'};
+    cursor: ${props => (props.isDisabled ? 'not-allowed' : 'pointer')};
+    color: #fff;
+    &:hover {
+        background-color: ${props => props.isDisabled && '#898990'};
+    }
+`;
+
+const ButtonClose = styled(Button)`
+    width: 18px;
+    height: 18px;
+    border: none;
+    position: absolute;
+    right: -10px;
+    top: -30px;
+
+    @media (max-width: 768px) {
+        top: 15px;
+        right: 15px;
+    }
+
+    path {
+        fill: ${colors.black};
+    }
+`;
+
+const StyledDialogOverlay = styled(DialogOverlay)`
+    &[data-reach-dialog-overlay] {
+        background-color: rgba(255, 255, 255, 0.8);
+        z-index: ${zIndex.overlay};
+
+        @media (min-width: 768px) {
+            overflow: hidden;
+        }
+    }
+`;
+
+const StyledDialogContent = styled(DialogContent)`
+    &[data-reach-dialog-content] {
+        position: relative;
+        width: 600px;
+        background: ${props => props.theme.lightBgColor};
+        border: 1px solid #e5e5e5;
+
+        @media (max-width: 768px) {
+            min-height: 100vh;
+            width: 100vw;
+            margin: 0;
+            padding: 30px;
+
+            border: none;
+        }
+    }
+`;
```

Comment:
Mostly copy-pasted from `change_pool_dialog`, it is small enough that it didn't feel worth it creating a generic dialog component but if we start adding more of these we should.

Reply:
that's my preferred level of abstraction too 😄